File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int count = 0;
    int majority = 0;

    for (int i = 0; i < numsSize; i++) {
        if (count == 0) {
            majority = nums[i];
        }
        count += (nums[i] == majority) ? 1 : -1;
    }

    return majority;
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hm = new HashMap<>();

        for (int n : nums) {
            hm.put(n, hm.getOrDefault(n, 0) + 1);
            if (hm.get(n) > (nums.length / 2))
                return n;
        }
        return 0;
    }
}